indexer/indexer_tests/editable_map_object_test.cpp

```diff
@@ -289,4 +289,461 @@ UNIT_TEST(EditableMapObject_SetInternet)
   setInternetAndCheck(bunkerEmo, osm::Internet::Wlan, true);
   setInternetAndCheck(bunkerEmo, osm::Internet::Wlan, true);
 }
+
+UNIT_TEST(EditableMapObject_RemoveFakeNames)
+{
+  EditableMapObject emo;
+  StringUtf8Multilang name;
+  osm::FakeNames fakeNames;
+
+  name.AddString(GetLangCode("default"), "Default name");
+  name.AddString(GetLangCode("ru"), "Default name");
+  name.AddString(GetLangCode("en"), "Default name");
+  fakeNames.names.push_back({GetLangCode("ru"), "Default name"});
+  fakeNames.names.push_back({GetLangCode("en"), "Default name"});
+  fakeNames.defaultName = "Default name";
+
+  EditableMapObject::RemoveFakeNames(fakeNames, name);
+
+  size_t counter = 0;
```

Comment:
Get rid of counter. Just check in the lambda that code is the default lang code.